pinecone_text/dense/cohere_encoder.py

```diff
@@ -0,0 +1,84 @@
+from enum import Enum
+from typing import Union, List, Any, Optional
+from pinecone_text.dense.base_dense_ecoder import BaseDenseEncoder
+
+try:
+    import cohere
+    from cohere.error import CohereError
+except (OSError, ImportError, ModuleNotFoundError) as e:
+    _cohere_installed = False
+else:
+    _cohere_installed = True
+
+
+class CohereInputType(Enum):
+    SEARCH_DOCUMENT = "search_document"
+    SEARCH_QUERY = "search_query"
+    # CLASSIFICATION = "classification"
+    # CLUSTERING = "clustering"
+
+
+class CohereEncoder(BaseDenseEncoder):
+    """
+    Cohere's text embedding wrapper. See https://docs.cohere.com/reference/embed
+
+    Note: You should provide an API key as the environment variable COHERE_API_KEY.
+          Or you can pass it as argument to the constructor as `api_key`.
+    """
+
+    def __init__(
+        self,
+        model_name: str = "embed-multilingual-v3.0",
+        api_key: Optional[str] = None,
+        **kwargs: Any,
+    ):
+        """
+        Initialize the Cohere encoder.
+
+        :param model_name: The name of the embedding model to use. See https://docs.cohere.com/reference/embed
+        :param kwargs: Additional arguments to pass to the underlying cohere client. See https://github.com/openai/openai-python
+        """
+        if not _cohere_installed:
+            raise ImportError(
+                "Failed to import cohere. Make sure you install cohere extra "
+                "dependencies by running: "
+                "`pip install pinecone-text[cohere]"
+            )
+        self._model_name = model_name
+        self._client = cohere.Client(api_key=api_key)
+
+    def encode_documents(
+        self, texts: Union[str, List[str]]
+    ) -> Union[List[float], List[List[float]]]:
+        return self._encode(texts, CohereInputType.SEARCH_DOCUMENT.value)
```

Comment:
What would happen if someone would use it with an older version than Cohere embeddings V3? Is their SDK knows to ignore the `input_type` property?